services/brig/src/Brig/Whitelist.hs

```diff
@@ -1,44 +1,68 @@
 -- | > docs/reference/user/activation.md {#RefActivationWhitelist}
 --
 -- Email/phone whitelist.
-module Brig.Whitelist (Whitelist (..), verify) where
+module Brig.Whitelist
+    ( Whitelist (..)
+    , InternalWhitelist (..)
+    , verifyInternal
+    , verifyService
+    ) where
 
 import Imports
 import Data.Aeson
+import Data.Aeson.TH (deriveJSON)
 import Bilge.IO
 import Bilge.Request
 import Bilge.Response
 import Bilge.Retry
 import Brig.Types
 import Control.Monad.Catch (MonadMask, throwM)
 import Control.Retry
-import Data.Text
+import qualified Data.Text as T
 import Data.Text.Encoding (encodeUtf8)
 import Network.HTTP.Client (HttpException (..), HttpExceptionContent (..), parseRequest)
+import Util.Options.Common (toOptionFieldName)
 
 -- | A service providing a whitelist of allowed email addresses and phone numbers
+-- DEPRECATED: use internal whitelisting instead!
 data Whitelist = Whitelist
     { whitelistUrl  :: !Text     -- ^ Service URL
-    , whitelistUser :: !Text     -- ^ Username
-    , whitelistPass :: !Text     -- ^ Password
+    , whitelistUser :: !Text     -- ^ Service Username (basic auth)
+    , whitelistPass :: !Text     -- ^ Service Password (basic auth)
+
     } deriving (Show, Generic)
 
 instance FromJSON Whitelist
 
+data InternalWhitelist = InternalWhitelist
+    { internalWhitelistDomains :: ![Text]} deriving (Show, Generic)
```

Comment:
Use newtype:

```
newtype InternalWhitelist = InternalWhitelist [Text]
```

Reply:
But that changes the config format from

```
setInternalWhitelist:
    whitelistDomains: [...]
```

into 

```
setInternalWhitelist: [...]
```

right?

Reply:
Correct. You could possibly rename the field in Options.hs to `setWhitelistDomains` (or something). This suggestion (using `newtype`) would remove one level of nesting. If you think you'll need other whitelist-related configurations other than the domains (I can't currently think of any), and you prefer deeply nested configs rather than configs where settings share a common prefix in their name, then feel free to leave this as-is.